proxy/envoy/discovery.go

```diff
@@ -302,6 +323,42 @@ func (ds *DiscoveryService) clearCache() {
 	ds.rdsCache.clear()
 }
 
+// ListAllEndpoints responds with all Services and is restricted to a single service key
+func (ds *DiscoveryService) ListAllEndpoints(request *restful.Request, response *restful.Response) {
+
+	servicesArray := make([]*service, 0)
+	for _, svc := range ds.services.Services() {
+		for _, port := range svc.Ports {
+
+			hostArray := make([]*host, 0)
+			for _, ep := range ds.services.Instances(svc.Hostname, []string{port.Name}, []model.Tags{}) {
+				hostArray = append(hostArray, &host{
+					Address: ep.Endpoint.Address,
+					Port:    ep.Endpoint.Port,
+				})
+			}
+
+			servicesArray = append(servicesArray, &service{
+				Key:   svc.Key(port, nil),
+				Hosts: hostArray,
+			})
+		}
+	}
+
+	// Sort servicesArray.  This is not strictly necessary, but discovery_test.go will
+	// be comparing against a golden example using test/util/diff.go which does a textual comparison
+	sort.Sort(serviceSorter(servicesArray))
```

Comment:
You can use [sort.Slice](https://golang.org/pkg/sort/#Slice) here instead of implementing sort.Interface